tableauserverclient/server/endpoint/projects_endpoint.py

```diff
@@ -50,3 +60,51 @@ def create(self, project_item):
         new_project = ProjectItem.from_response(server_response.content, self.parent_srv.namespace)[0]
         logger.info('Created new project (ID: {0})'.format(new_project.id))
         return new_project
+
+    @api(version='2.0')
+    def populate_permissions(self, item):
+        self._permissions.populate(item)
+
+    @api(version='2.0')
+    def update_permission(self, item, rules):
+        return self._permissions.update(item, rules)
+
+    @api(version='2.0')
+    def delete_permission(self, item, rules):
+        return self._permissions.delete(item, rules)
+
+    @api(version='2.1')
+    def populate_workbook_default_permissions(self, item):
```

Comment:
I might be remembering wrong, but I thought we talked about having 1 method for each verb that takes in a map containing all default permissions for all resource types?

Reply:
We did at first, but when we reconsidered in terms of type-safety (or at least the premise of it with autocomplete and maybe py3 mypy stuff that I'd love ot add) it felt a little safer to go this route, than having to introspect dicts and such.

Basically, is it less work to say
``` server.update(THING_TYPE, THING)```
or
```server.update_type_of_thing(thing)```

And I think for consistency and simplicity, the latter won out.